src/tree.rs

```diff
@@ -14,11 +14,14 @@ pub struct MerkleProof {
     pub path: Vec<Node>,
 }
 
+// `NonMembershipProof` contains the `MerkleProof` of the node where the returned `missing_node: LeafNode` would be found.
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct NonMembershipProof {
-    // Root hash of the Merkle Tree.
+    // Merkle proof of the node that `missing_node` would be found between `label` and `next`.
     pub merkle_proof: MerkleProof,
-    // Path from the leaf to the root.
+    // Index of the node that `missing_node` would be found between `label` and `next`.
+    pub closest_index: usize,
```

Comment:
So far, all Merkle proofs start with the leaf (i.e. path[0]) for which the proof is created. This is also the case here and we have a function to find the index of a leaf. Do we need this then? Probably for efficiency reasons...